openstack/networking/v2/networks/results.go

```diff
@@ -93,6 +93,10 @@ type Network struct {
 
 	// RevisionNumber optionally set via extensions/standard-attr-revisions
 	RevisionNumber int `json:"revision_number"`
+
+	// IsExternalNetwork reports whether this is an external network or not (typically a
+	// network providing router and floating IP subnets etc).  
+	IsExternalNetwork bool `json:"router:external"`
```

Comment:
it's already implemented. you need to use https://pkg.go.dev/github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/external

Reply:
Yeah, just discovered this via the CI failure!  Can confirm it does work with my app.